File: dynamicpdf-csharp-examples/Examples/PieChartsExample.cs

using ceTe.DynamicPDF;
using ceTe.DynamicPDF.PageElements.Charting;
using ceTe.DynamicPDF.PageElements.Charting.Series;

namespace DynamicPDFCoreSuite.Examples
{
    public class PieChartsExample
    {
        public static void Run()
        {
            SimplePlotAreaExample();
            SimplePlotAreaPercentageExample();
            ManualExample();
        }


        public static void SimplePlotAreaExample()
        {
            Document doc = new();
            doc.Pages.Add(new Page(PageSize.A4, PageOrientation.Landscape));

            Chart chart = new Chart(0, 0, 700, 400);

            Title tTitle = new Title("Website Viewers (in millions)");
            Title tTitle1 = new Title("Year 2007");
            chart.HeaderTitles.Add(tTitle);
            chart.HeaderTitles.Add(tTitle1);

            
            ScalarDataLabel da = new ScalarDataLabel(true, false, false);
            PieSeries pieSeries = new PieSeries();
            pieSeries.DataLabel = da;
            chart.PrimaryPlotArea.Series.Add(pieSeries);

            pieSeries.Elements.Add(27, "Website A");
            pieSeries.Elements.Add(19, "Website B");
            pieSeries.Elements.Add(21, "Website C");

            pieSeries.Elements[0].Color = new AutoGradient(90f, CmykColor.Red, CmykColor.IndianRed);
            pieSeries.Elements[1].Color = new AutoGradient(90f, CmykColor.Green, CmykColor.YellowGreen);
            pieSeries.Elements[2].Color = new AutoGradient(90f, CmykColor.Blue, CmykColor.LightBlue);


            doc.Pages[0].Elements.Add(chart);
            doc.Draw(Util.GetPath("Output/piechart-simple-output.pdf"));
        }

        public static void ManualExample()
        {
            string outputPath = Util.GetPath("Output/piechart-manual-output.pdf");
            Document document = new();
            document.Pages.Add(new Page());

            Chart chart = new Chart(0, 0, 500, 500);
            chart.BackgroundColor = RgbColor.LightBlue;
            chart.AutoLayout = false;

            PlotArea plotArea = chart.PlotAreas.Add(10, 100, 100, 100);
            plotArea.BackgroundColor = RgbColor.Tan;
            PieSeries pieSeries = new PieSeries();
            ScalarDataLabel dataLbl = new ScalarDataLabel(true, false, false);
            pieSeries.DataLabel = dataLbl;

            plotArea.Series.Add(pieSeries);
            PieSeriesElement pe1 = new(10, "A", RgbColor.Green);
            PieSeriesElement pe2 = new(20, "B", RgbColor.Red);
            PieSeriesElement pe3 = new(13, "C", RgbColor.Purple);

            pieSeries.Elements.Add(pe1);
            pieSeries.Elements.Add(pe2);
            pieSeries.Elements.Add(pe3);

            
            chart.Layout();

            chart.Legends[0].X = 400;
            chart.Legends[0].Y = 250;


            document.Pages[0].Elements.Add(chart);
            document.Draw(outputPath);
        }


        public static void SimplePlotAreaPercentageExample()
        {
            Document doc = new();
            doc.Pages.Add(new Page(PageSize.A4, PageOrientation.Landscape));

            Chart chart = new Chart(0, 0, 700, 400);

            Title tTitle = new Title("Website Viewers (in millions)");
            Title tTitle1 = new Title("Year 2007");
            chart.HeaderTitles.Add(tTitle);
            chart.HeaderTitles.Add(tTitle1);


            ScalarDataLabel da = new ScalarDataLabel(true, true, true);
            PieSeries pieSeries = new PieSeries();
            pieSeries.DataLabel = da;
            chart.PrimaryPlotArea.Series.Add(pieSeries);

            pieSeries.Elements.Add(27, "Website A");
            pieSeries.Elements.Add(19, "Website B");
            pieSeries.Elements.Add(21, "Website C");

            pieSeries.Elements[0].Color = new AutoGradient(90f, CmykColor.Red, CmykColor.IndianRed);
            pieSeries.Elements[1].Color = new AutoGradient(90f, CmykColor.Green, CmykColor.YellowGreen);
            pieSeries.Elements[2].Color = new AutoGradient(90f, CmykColor.Blue, CmykColor.LightBlue);


            doc.Pages[0].Elements.Add(chart);
            doc.Draw(Util.GetPath("Output/piechart-simple-percentage-output.pdf"));
        }


    }
}

File: dynamicpdf-core-suite-vb-core/Examples/PieChartsExample.vb
Imports System.IO
Imports ceTe.DynamicPDF
Imports ceTe.DynamicPDF.PageElements.Charting
Imports ceTe.DynamicPDF.PageElements.Charting.Series

Public Class PieChartsExample

    Public Shared Sub Run()

        Dim outputPath As String
        outputPath = Util.GetPath("Output/pie-chart-output.pdf")

        ' Create a PDF Document
        Dim MyDocument As New Document()

        ' Create a Page and add it to the MyDocument
        Dim MyPage As New Page(PageSize.A4, PageOrientation.Landscape)
        MyDocument.Pages.Add(MyPage)

        ' Create a MyChart
        Dim MyChart = New Chart(0, 0, 700, 400)
        ' Add a plot area to the MyChart
        Dim MyPlotArea As PlotArea = MyChart.PlotAreas.Add(50, 50, 300, 300)

        ' Create the Header titles and add it to the MyChart
        Dim MytTitle As New Title("Website Viewers (in millions)")
        Dim MytTitle1 As New Title("Year 2007")
        MyChart.HeaderTitles.Add(MytTitle)
        MyChart.HeaderTitles.Add(MytTitle1)

        ' Create autogradient colors 
        Dim MyAutogradient1 As AutoGradient = New AutoGradient(90.0F, CmykColor.Red, CmykColor.IndianRed)
        Dim MyAutogradient2 As AutoGradient = New AutoGradient(90.0F, CmykColor.Green, CmykColor.YellowGreen)
        Dim MyAutogradient3 As AutoGradient = New AutoGradient(90.0F, CmykColor.Blue, CmykColor.LightBlue)

        ' Create a scalar datalabel
        Dim Myda As ScalarDataLabel = New ScalarDataLabel(True, False, False)
        ' Create a pie series 
        Dim MyPieSeries As New PieSeries()
        ' Set scalar datalabel to the pie series
        MyPieSeries.DataLabel = Myda
        ' Add series to the plot area
        MyPlotArea.Series.Add(MyPieSeries)

        'Add pie series elements to the pie series
        MyPieSeries.Elements.Add(27, "Website A")
        MyPieSeries.Elements.Add(19, "Website B")
        MyPieSeries.Elements.Add(21, "Website C")

        ' Assign autogradient colors to series elements
        MyPieSeries.Elements(0).Color = MyAutogradient1
        MyPieSeries.Elements(1).Color = MyAutogradient2
        MyPieSeries.Elements(2).Color = MyAutogradient3

        ' Add the MyChart to the MyPage
        MyPage.Elements.Add(MyChart)
        ' Save the pdf
        MyDocument.Draw(outputPath)
    End Sub


End Class
